crates/circuit/src/operations.rs

```
@@ -222,7 +224,9 @@ impl Param {
             _ => DEEPCOPY
                 .get_bound(py)
                 .call1((self.clone(), memo))?
-                .extract(),
+                .extract()
+                // The extraction is infallible.
+                .map_err(|x| match x {}),
```

Is this a normal pattern for discarding the error here? I probably have done something like:

```rust
Ok(extract().unwrap())
```

or something like that

---

I did my way so the `unwrap`/`expect` will become a compilation error (unhandled match variants) if we change the error type and forget to update the handling.  In nightly there's a `Result::into_ok` that would be clearer, but no date yet for a release of that.

---

> Is this a normal pattern for discarding the error here?

Just to be clear, this doesn't _discard_ the error - it completely handles all variants.  It's just that the error is uninhabitable.  If the error were any type (even `()`) then my code would be a compile-time error.